transaction/core/src/blockchain/block_metadata.rs

```diff
@@ -0,0 +1,124 @@
+
+use crate::{BlockID, QuorumSet};
+use displaydoc::Display;
+use mc_crypto_digestible::{Digestible, MerlinTranscript};
+use mc_crypto_keys::{
+    Ed25519Pair, Ed25519Public, Ed25519Signature, SignatureError, Signer, Verifier,
+};
+use prost::Message;
+use serde::{Deserialize, Serialize};
+
+/// Metadata for a block.
+#[derive(Clone, Deserialize, Digestible, Display, Eq, Message, PartialEq, Serialize)]
+pub struct BlockMetadata {
+    /// The Block ID.
+    #[prost(message, required, tag = 1)]
+    block_id: BlockID,
+
+    /// Quorum set configuration at the time of externalization.
+    #[prost(message, optional, tag = 2)]
+    quorum_set: Option<QuorumSet>,
+
+    /// IAS report for the enclave which generated the signature.
+    #[prost(message, optional, tag = 3)]
+    verification_report: Option<VerificationReport>,
+}
+
+impl BlockMetadata {
+    /// Instantiate a [BlockMetadata] with the given data.
+    pub fn new(
+        block_id: BlockID,
+        quorum_set: Option<QuorumSet>,
+        verification_report: Option<VerificationReport>,
+    ) -> Self {
+        Self {
+            block_id,
+            quorum_set,
+            verification_report,
+        }
+    }
+
+    /// Get the [BlockID].
+    pub fn block_id(&self) -> &BlockID {
+        &self.block_id
+    }
+
+    /// Get the [QuorumSet].
+    pub fn quorum_set(&self) -> &Option<QuorumSet> {
+        &self.quorum_set
+    }
+
+    /// Get the Attested [VerificationReport].
+    pub fn verification_report(&self) -> &Option<VerificationReport> {
+        &self.verification_report
+    }
+
+    fn digest(&self) -> [u8; 32] {
+        self.digest32::<MerlinTranscript>(b"block_metadata")
+    }
+}
+
+/// Signed metadata for a block.
+#[derive(Clone, Deserialize, Digestible, Display, Eq, Message, PartialEq, Serialize)]
+pub struct SignedBlockMetadata {
+    /// Metadata signed by the consensus node.
+    #[prost(message, required, tag = 1)]
+    contents: BlockMetadata,
+
+    /// Message signing key (signer).
+    #[prost(message, required, tag = 2)]
+    node_key: Ed25519Public,
+
+    /// Signature using `node_key` over the Digestible encoding of `contents`.
+    #[prost(message, required, tag = 3)]
+    digest_signature: Ed25519Signature,
+}
+
+impl SignedBlockMetadata {
+    /// Instantiate a [SignedBlockMetadata] with the given data.
+    pub fn new(
+        contents: BlockMetadata,
+        node_key: Ed25519Public,
+        digest_signature: Ed25519Signature,
+    ) -> Self {
+        Self {
+            contents,
+            node_key,
+            digest_signature,
+        }
+    }
+
+    /// Instantiate a [SignedBlockMetadata] by signing the given [BlockMetadata]
+    /// with the given [Ed25519Pair].
+    pub fn from_contents_and_keypair(
+        contents: BlockMetadata,
+        key_pair: &Ed25519Pair,
+    ) -> Result<Self, SignatureError> {
+        let signature = key_pair.try_sign(&contents.digest())?;
```

Comment:
_Originally posted by @iamalwaysuncomfortable in https://github.com/mobilecoinfoundation/mobilecoin/pull/2052#discussion_r881897742_

> I'm unsure whether we should be using `try_sign_digest` or `try_sign` here. 
>
> Specifically:
> - `digest()` makes a hash of a `message` and I'm unsure whether the `digest` should be considered the `message` itself or whether the `digest` should be considered a `pre-hash` of the message? Perhaps we should be signing a specific byte representation of `BlockMetaData` first? Any advice from @isis-mc or @garbageslam would be welcome here.
>- I'm also unsure whether we should have any domain separators added to the digest first. I think we might want that?

Reply:
My intuition here is the digest is the correct message (it'd go through another hash, but that's a known set of functions/morphisms to apply) - but maybe a digest with a domain separator. My other intuition is that we might do what's already being done around the codebase as far as "what constitutes the `message` being signed" - but I havne't looked deep enough to know exactly what that is.

Reply:
Tagging @isis-mc for a review on this. Thank you

Reply:
Switched to using `DigestibleSigner`/`DigestibleVerifier`, per this comment and discussion with James. IIUC it's the same operations, with slightly different structure. Definitely could use Isis' review.